ThreadSocket/server.c

```diff
@@ -0,0 +1,69 @@
+#include <arpa/inet.h>
+#include <fcntl.h>
+#include <netinet/in.h>
+#include <pthread.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/socket.h>
+#include <unistd.h>
+char client_message[2000];
+char buffer[1024] = "Message from server";
+pthread_mutex_t lock = PTHREAD_MUTEX_INITIALIZER;
+void *SocketThread(void *arg) 
+{
+  int newSocket = *((int *)arg);
+  recv(newSocket, client_message, 2000, 0);
+  printf("%s\n", client_message);
+  pthread_mutex_lock(&lock);
```

Comment:
Why do we need to lock here? Can you explain?
When will you release the lock? I don't see any "pthread_mutex_unlock" is called ?

Reply:
pthread_mutex_lock and pthread_mutex_unlock make sure that in anytime just only one thread access to critical resource 
That is what I found out about mutex  and I updated  mistake code